api/src/routes/edit.ts

```diff
@@ -82,6 +89,37 @@ async function getChildren(req, res) {
     res.json(response);
 }
 
+function uploadFile(req, res, next) {
+    const { pid } = req.params;
+    const form = formidable({ multiples: true });
+
+    form.parse(req, async (err, fields, files) => {
+        if (err) {
+            next(err);
+            return;
+        }
+        try {
+            const datastream = Datastream.getInstance();
+            const { stream } = fields;
+            const { filepath, mimetype } = files?.file;
+            await datastream.uploadFile(pid, stream, filepath, mimetype);
+            res.status(200).send("Upload success");
+        } catch (error) {
+            res.status(500).send(error.message);
+        }
+    });
+}
+
+edit.get("/object/details/:pid", requireToken, pidSanitizer, async function (req, res) {
+    try {
+        const data = await HierarchyCollector.getInstance().getFedoraData(req.params.pid);
+        res.json({ models: data.models, datastreams: data.fedoraDatastreams });
+    } catch (error) {
+        console.error(error);
+        res.status(500).send(error.message);
+    }
+});
+edit.post("/object/:pid", requireToken, pidSanitizer, uploadFile);
```

Comment:
Maybe we should use a more specific path here, like `/object/datastreamUpload/:pid`. In theory, `:pid` can be practically anything (though in practice, we constrain it). I want to be sure we avoid ambiguity in our URL pathing.

I should also say that I'm not very happy with the current `/object/[action]/[pid]` pattern, when I think `/object/[pid]/action` would make a lot more sense. It just got started along this road because of the way we set up `/object/children` to get all top-level objects in the hierarchy, with `/object/children/:pid` to get children of a specific object. With that precedent set, everything else fell into place... but I'd really like to change it. If we rename `/object/children` to something like `/hierarchy/top` then the rest could probably be flipped around... but there have always been so many pull requests in progress at once, there hasn't been a good opportunity to address this in a single place.